statusHandler/persister/persistentHandler.go

```diff
@@ -0,0 +1,200 @@
+	startSaveInStorage       bool
+}
+
+// NewPersistentStatusHandler will return an instance of the persistent status handler
+func NewPersistentStatusHandler(
+	marshalizer marshal.Marshalizer,
+	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter,
+) (*PersistentStatusHandler, error) {
+	if check.IfNil(marshalizer) {
+		return nil, process.ErrNilMarshalizer
+	}
+	if check.IfNil(uint64ByteSliceConverter) {
+		return nil, process.ErrNilUint64Converter
+	}
+
+	psh := new(PersistentStatusHandler)
+	psh.store = storageUnit.NewNilStorer()
+	psh.uint64ByteSliceConverter = uint64ByteSliceConverter
+	psh.marshalizer = marshalizer
+	psh.persistentMetrics = &sync.Map{}
+	psh.initMap()
+
+	go func() {
+		time.Sleep(time.Second)
+
+		psh.startSaveInStorage = true
+	}()
+
+	return psh, nil
+}
+
+func (psh *PersistentStatusHandler) initMap() {
+	initUint := uint64(0)
+
+	psh.persistentMetrics.Store(core.MetricCountConsensus, initUint)
+	psh.persistentMetrics.Store(core.MetricCountConsensusAcceptedBlocks, initUint)
+	psh.persistentMetrics.Store(core.MetricCountAcceptedBlocks, initUint)
+	psh.persistentMetrics.Store(core.MetricCountLeader, initUint)
+	psh.persistentMetrics.Store(core.MetricNumProcessedTxs, initUint)
+	psh.persistentMetrics.Store(core.MetricNumShardHeadersProcessed, initUint)
+	psh.persistentMetrics.Store(core.MetricNonce, initUint)
+}
+
+// SetStorage will set storage for persistent status handler
+func (psh *PersistentStatusHandler) SetStorage(store storage.Storer) error {
+	if check.IfNil(store) {
+		return process.ErrNilStorage
+	}
+
+	psh.store = store
+
+	return nil
+}
+
+func (psh *PersistentStatusHandler) saveMetricsInDb(nonce uint64) {
+	metricsMap := make(map[string]interface{})
+	psh.persistentMetrics.Range(func(key, value interface{}) bool {
+		keyString, ok := key.(string)
+		if !ok {
+			return false
+		}
+		metricsMap[keyString] = value
+		return true
+	})
+
+	statusMetricsBytes, err := psh.marshalizer.Marshal(&metricsMap)
+	if err != nil {
+		log.Debug("cannot marshal metrics map",
+			"error", err)
+		return
+	}
+
+	nonceBytes := psh.uint64ByteSliceConverter.ToByteSlice(nonce)
+	err = psh.store.Put(nonceBytes, statusMetricsBytes)
+	if err != nil {
+		log.Debug("cannot save metrics map in storage",
+			"error", err)
+		return
+	}
+}
+
+// SetInt64Value method - will update the value for a key
+func (psh *PersistentStatusHandler) SetInt64Value(key string, value int64) {
+	if _, ok := psh.persistentMetrics.Load(key); !ok {
+		return
+	}
+
+	psh.persistentMetrics.Store(key, value)
+}
+
+// SetUInt64Value method - will update the value for a key
+func (psh *PersistentStatusHandler) SetUInt64Value(key string, value uint64) {
+	valueFromMapI, ok := psh.persistentMetrics.Load(key)
+	if !ok {
+		return
+	}
+
+	psh.persistentMetrics.Store(key, value)
+
+	//metrics wil be saved in storage every time when a block is committed successfully
+	if key != core.MetricNonce {
```

Comment:
Why this metric is not saved in this storage?

Reply:
nonce is saved in storage. And this condition is needed to trigger metrics to be saved in storage.

Reply:
Ok, actually the question was why the method SetUint64Value does not save any other metrics beside MetricNonce